### ISSUE TITLE
[Windows] STATUS_ACCESS_VIOLATION: Segmentation fault

### ISSUE THREAD

**Issue by notdanilo** [PoloDB/PoloDB#166]
On Windows 11, the following code fails with

```
error: process didn't exit successfully: target\debug\rust-example.exe (exit code: 0xc0000005, STATUS_ACCESS_VIOLATION)
Segmentation fault
```

```rust
use polodb_core::{CollectionT, Database};
use serde::{Serialize, Deserialize};

#[derive(Debug, Serialize, Deserialize)]
struct Book {
    title: String,
    author: String,
}

fn main() -> polodb_core::Result<()> {
    let db = Database::open_path("./data.db")?;
    let collection = db.collection::<Book>("books");
    collection.insert_one(Book {
        title: "The Three-Body Problem".to_string(),
        author: "Liu Cixin".to_string(),
    })?;    
    Ok(())
}
```

while this works

```rust
fn main() {
  std::fs::write("./data.db", "hello").unwrap();
}
```

**Comment by vincentdchan** [comment]
Please make sure that you compiles PoloDB with MSVC's clang(not MingW). Make sure install Clang in VS
![CleanShot 2024-09-03 at 10 22 15@2x](https://github.com/user-attachments/assets/2d9bd06a-3d9e-434e-ad64-e019fbf255ca)
And set the env to the correct path: `$Env:LIBCLANG_PATH="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\Llvm\bin"`

**Comment by notdanilo** [comment]
```c++
_Thrd_yield (@_Thrd_yield:83)
public: void __cdecl std::_Mutex_base::lock(void) (c:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\mutex:52)
public: void __cdecl rocksdb::port::Mutex::Lock(void) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\port\win\port_win.h:95)
public: __cdecl rocksdb::MutexLock::MutexLock(class rocksdb::port::Mutex *) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\util\mutexlock.h:37)
public: virtual unsigned __int64 __cdecl rocksdb::ShardedCacheBase::GetCapacity(void)const  (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\cache\sharded_cache.cc:82)
protected: unsigned __int64 __cdecl rocksdb::ShardedCacheBase::GetPerShardCapacity(void)const  (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\cache\sharded_cache.cc:74)
public: __cdecl rocksdb::lru_cache::LRUCache::LRUCache(struct rocksdb::LRUCacheOptions const &) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\cache\lru_cache.cc:651)
void std::_Construct_in_place<rocksdb::lru_cache::LRUCache,rocksdb::LRUCacheOptions &>(class rocksdb::lru_cache::LRUCache & const, struct rocksdb::LRUCacheOptions & const) (c:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\xutility:403)
void std::_Ref_count_obj2<rocksdb::lru_cache::LRUCache>::_Ref_count_obj2<rocksdb::lru_cache::LRUCache><rocksdb::LRUCacheOptions &>(struct rocksdb::LRUCacheOptions & const) (c:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\memory:2095)
class std::shared_ptr<rocksdb::lru_cache::LRUCache> std::make_shared<rocksdb::lru_cache::LRUCache,rocksdb::LRUCacheOptions &>(struct rocksdb::LRUCacheOptions & const) (c:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\memory:2906)
public: class std::shared_ptr<class rocksdb::Cache> __cdecl rocksdb::LRUCacheOptions::MakeSharedCache(void)const  (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\cache\lru_cache.cc:709)
class std::shared_ptr<class rocksdb::Cache> __cdecl rocksdb::NewLRUCache(struct rocksdb::LRUCacheOptions const &) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\include\rocksdb\cache.h:287)
protected: void __cdecl rocksdb::BlockBasedTableFactory::InitializeOptions(void) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\table\block_based\block_based_table_factory.cc:449)
public: __cdecl rocksdb::BlockBasedTableFactory::BlockBasedTableFactory(struct rocksdb::BlockBasedTableOptions const &) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\table\block_based\block_based_table_factory.cc:422)
public: __cdecl rocksdb::ColumnFamilyOptions::ColumnFamilyOptions(void) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\options\options.cc:123)
public: __cdecl rocksdb::Options::Options(void) (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\include\rocksdb\options.h:1459)
```

```rust
rocksdb_options_t::rocksdb_options_t(void) (@rocksdb_options_t::rocksdb_options_t(void):8)
rocksdb_options_create (d:\dev\3rd\PoloDB\src\librocksdb-sys\rocksdb\db\c.cc:2804)
static union enum2$<core::result::Result<polodb_core::db::rocksdb_wrapper::RocksDBWrapperInner,enum2$<polodb_core::errors::Error> > > polodb_core::db::rocksdb_wrapper::RocksDBWrapperInner::open(struct ref$<std::path::Path>) (d:\dev\3rd\PoloDB\src\polodb_core\db\rocksdb_wrapper.rs:80)
union enum2$<core::result::Result<polodb_core::db::rocksdb_wrapper::RocksDBWrapper,enum2$<polodb_core::errors::Error> > > polodb_core::db::rocksdb_wrapper::RocksDBWrapper::open(struct ref$<std::path::Path>) (d:\dev\3rd\PoloDB\src\polodb_core\db\rocksdb_wrapper.rs:49)
```

Stop reason: Exception 0xc0000005 encountered at address 0x7ffb20a53020: Access violation reading location 0x00000000

It's apparently trying to read a null ptr

**Comment by notdanilo** [comment]
This issue is probably related to https://github.com/rust-rocksdb/rust-rocksdb/issues/665

**Comment by vincentdchan** [comment]
Fixed by #167 
Published as `v5.1.1`
